impeller/display_list/dl_aiks_canvas.cc

```diff
@@ -0,0 +1,1191 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/display_list/dl_aiks_canvas.h"
+
+#include <algorithm>
+#include <cstring>
+#include <memory>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/trace_event.h"
+#include "impeller/aiks/color_filter.h"
+#include "impeller/core/formats.h"
+#include "impeller/display_list/dl_image_impeller.h"
+#include "impeller/display_list/dl_vertices_geometry.h"
+#include "impeller/display_list/nine_patch_converter.h"
+#include "impeller/display_list/skia_conversions.h"
+#include "impeller/entity/contents/conical_gradient_contents.h"
+#include "impeller/entity/contents/filters/filter_contents.h"
+#include "impeller/entity/contents/filters/inputs/filter_input.h"
+#include "impeller/entity/contents/linear_gradient_contents.h"
+#include "impeller/entity/contents/radial_gradient_contents.h"
+#include "impeller/entity/contents/runtime_effect_contents.h"
+#include "impeller/entity/contents/scene_contents.h"
+#include "impeller/entity/contents/sweep_gradient_contents.h"
+#include "impeller/entity/contents/tiled_texture_contents.h"
+#include "impeller/entity/entity.h"
+#include "impeller/entity/geometry/geometry.h"
+#include "impeller/geometry/path.h"
+#include "impeller/geometry/path_builder.h"
+#include "impeller/geometry/scalar.h"
+#include "impeller/geometry/sigma.h"
+#include "impeller/typographer/backends/skia/text_frame_skia.h"
+
+namespace impeller {
+
+#define UNIMPLEMENTED \
+  FML_DLOG(ERROR) << "Unimplemented detail in " << __FUNCTION__;
+
+DlAiksCanvas::DlAiksCanvas(const SkRect& cull_rect, bool prepare_rtree)
+    : canvas_(skia_conversions::ToRect(cull_rect)) {
+  if (prepare_rtree) {
+    accumulator_ = std::make_unique<flutter::RTreeBoundsAccumulator>();
+  }
```

Comment:
else - a rect accumulator. We usually set the flag on most internal uses, but if anyone ever constructs the object with the flag set to false, it will just crash on the first Save call.